testsuite/classlibrary/TestAbstractFunction.sc

```diff
@@ -0,0 +1,137 @@
+TestAbstractFunction : UnitTest {
+
+	test_rest_binop {
+		var func;
+		var args = [[1, Rest(1)], [Rest(1), 1], [Rest(1), Rest(1)], [1, Rest()], [Rest(), Rest()]];
+		var funcs = [
+			{ |a, b| a * b },
+			{ |a, b| a + b },
+			{ |a, b| a - b },
+			{ |a, b| a / b }
+		];
+		var all = args.collect { |pair|
+			funcs.collect { |f| f.valueArray(pair) }
+		};
+		all.postln;
+
+		this.assert(
+			all.flat.every(_.isKindOf(Rest)),
+			"All binary math operations on Rest should return Rest.\n"
+		);
+
+		this.assert(
+			all.as(Set).postln.size == 1,
+			"All binary math operations on Rest should have equal value for equivalent instance values. Default value should be 1\n"
+		);
+
+		func = { |op, a, b| a.perform(op, b) == a.value.perform(op, b.value) };
+		all = [(0..2), (0..2).collect(Rest(_))].allTuples;
+		all = all ++ all.collect(_.reverse);
+		all.postln;
+		this.assert(
+			['<', '>', '<=', '>='].every { |op| all.every { |args| func.(op, *args) } },
+			"Instances of Rest should return a boolean for ordering operators"
+		);
+
+
+	}
+
+	test_rest_event {
+
+		this.assert(Rest() == Rest(1), "Rest default value should be 1");
+
+		this.assert(
+			(degree: Rest(0), parent: Event.default.parent).use { ~freq.value } == Rest(60.midicps),
+			"rest passes through event calculations"
+		);
+
+		this.assert(
+			(degree: Rest(7), octave: Rest(4), parent: Event.default.parent).delta == 1,
+			"delta is not touched by venet calculations"
```

Comment:
Typo; I wouldn't point it out normally but it's actually hard to tell that this was meant to say 'event'